### ISSUE TITLE
Dotnet scaffold fails without error when Default Schema is missing.

### ISSUE THREAD

**Issue by youss2017** [dotnet/efcore#30845]
### Problem:
When scaffolding a database without a Default-Schema in SQL Server, the dotnet ef process will hang infinitely without showing an error. This could take days to resolve and determine the underlying issue with dotnet scaffolding.
### Steps to reproduce:
1) Create a test database with a test table.
2) Ensure the database does not have the Default-Schema set.
3) Run dotnet scaffolding; example:
dotnet ef dbcontext scaffold $Connection Microsoft.EntityFrameworkCore.SqlServer -o Model -f --no-build --verbose --force -d -c "TestContext"

**Note**
To remove Default-Schema open security window dialog (for the user you're using to login) and remove dbo from Default-Scehma
![default_scehma](https://user-images.githubusercontent.com/30071252/236853341-77fb9a03-e615-45b7-b76d-16456e8da8a7.png)

Also, if you go to TestDb->Tasks->Generate Scripts you will get the following error in Microsoft SQL Server Managment Studio
![Screenshot 2023-05-08 103006](https://user-images.githubusercontent.com/30071252/236853038-48f4b903-11d0-4d52-b046-70f4f86513d8.png)

### Solution:
Show an error while scaffolding that the database does not have a Default-Schema to help the user solve the underlying issue.

**Comment by ErikEJ** [comment]
I am not able to repro the schema removal, can you share the generated SQL script to do this?

**Comment by ajcvickers** [comment]
@youss2017 Why do you need a database without a default schema?

**Comment by youss2017** [comment]
I do not need a database without a default-schema, I made a mistake when generating the database. An error message would be useful to future developers.

**Comment by youss2017** [comment]
Use the following SQL to create a test database
`

CREATE DATABASE TestDb;
GO
USE TestDb;
GO

CREATE LOGIN NormalUser WITH PASSWORD='Pa$$word';
CREATE LOGIN BugUser WITH PASSWORD='Pa$$word';
GO
CREATE USER NormalUser FOR LOGIN NormalUser WITH DEFAULT_SCHEMA=[dbo];
CREATE USER BugUser FOR LOGIN BugUser WITH DEFAULT_SCHEMA=[wrongSchema];

ALTER ROLE [db_owner] ADD MEMBER NormalUser;
ALTER ROLE [db_owner] ADD MEMBER BugUser;

CREATE TABLE TestTable (
	ID int not null primary key
);
`
The following commands to scaffold
dotnet ef dbcontext scaffold "Server=localhost;Database=TestDb;User Id=NormalUser;Password=Pa$$word;Initial Catalog=TestDb;" Microsoft.EntityFrameworkCore.SqlServer -o Data/Model -f --no-build --verbose --force -d -c "TestContext"

dotnet ef dbcontext scaffold "Server=localhost;Database=TestDb;User Id=BugUser;Password=Pa$$word;Initial Catalog=TestDb;" Microsoft.EntityFrameworkCore.SqlServer -o Data/Model -f --no-build --verbose --force -d -c "TestContext"

**Comment by ajcvickers** [comment]
Note for triage: I just spent an hour attempting to create a user without schema; I failed. Trying to clear it in SMS doesn't work for me--it just gets reset back to its previous value. I also can't see how this could hang unless there is an underlying issue in SqlClient or SQL Server, since we if an exception is thrown or if the value returned by GetDefaultSchema is not valid, then that should escape to the shell as an unhandled exception. So I suspect that SqlClient/SQL Server is hanging.
